lib/buy/onramper/onramper_buy_provider.dart

```diff
@@ -31,74 +43,330 @@ class OnRamperBuyProvider extends BuyProvider {
+
+      if (response.statusCode == 200) {
+        final Map<String, dynamic> data = jsonDecode(response.body) as Map<String, dynamic>;
+
+        final List<dynamic> onramps = data['message'] as List<dynamic>;
+
+        final Map<String, dynamic> result = {
+          for (var onramp in onramps)
+            (onramp['id'] as String): {
+              'displayName': onramp['displayName'] as String,
+              'svg': onramp['icons']['svg'] as String
+            }
+        };
+
+        return result;
+      } else {
+        print('Failed to fetch onramp metadata');
+        return {};
+      }
+    } catch (e) {
+      print('Error occurred: $e');
+      return {};
     }
   }
 
-  String getColorStr(Color color) {
-    return color.value.toRadixString(16).replaceAll(RegExp(r'^ff'), "");
+  @override
+  Future<List<Quote>?> fetchQuote(
+      {required Currency sourceCurrency,
+      required Currency destinationCurrency,
+      required double amount,
+      required bool isBuyAction,
+      required String walletAddress,
+      PaymentType? paymentType,
+      String? countryCode}) async {
+    String? paymentMethod;
+
+    if (paymentType != null && paymentType != PaymentType.all) {
+      paymentMethod = normalizePaymentMethod(paymentType);
+      if (paymentMethod == null) paymentMethod = paymentType.name;
+    }
+
+    final actionType = isBuyAction ? 'buy' : 'sell';
+
+    final normalizedSourceCurrency = _getNormalizeCryptoCurrency(sourceCurrency);
+    final normalizedDestinationCurrency = _getNormalizeCryptoCurrency(destinationCurrency);
+
+    final params = {
+      'amount': amount.toString(),
+      if (paymentMethod != null) 'paymentMethod': paymentMethod,
+      'uuid': 'acad3928-556f-48a1-a478-4e2ec76700cd',
+      'clientName': 'CakeWallet',
+      'type': actionType,
+      'walletAddress': walletAddress,
+      'isRecurringPayment': 'false',
+      'input': 'source',
+    };
+
+    log('Onramper: Fetching $actionType quote: $normalizedSourceCurrency -> $normalizedDestinationCurrency, amount: $amount, paymentMethod: $paymentMethod');
+
+    final url = Uri.https(
+        _baseApiUrl, '$quotes/$normalizedSourceCurrency/$normalizedDestinationCurrency', params);
+    final headers = {'Authorization': _apiKey, 'accept': 'application/json'};
+
+    try {
+      final response = await http.get(url, headers: headers);
+
+      if (response.statusCode == 200) {
+        final data = jsonDecode(response.body) as List<dynamic>;
+        if (data.isEmpty) return null;
+
+        List<Quote> validQuotes = [];
+
+        final onrampMetadata = await getOnrampMetadata();
+
+        for (var item in data) {
+          if (item['errors'] != null) break;
+
+          final paymentMethod = (item as Map<String, dynamic>)['paymentMethod'] as String;
+
+          final rampId = item['ramp'] as String?;
+          final rampMetaData = onrampMetadata[rampId] as Map<String, dynamic>?;
+
+          if (rampMetaData == null) break;
```

Comment:
why break? why not just continue to the next item(onramp)?
```suggestion
          if (rampMetaData == null) continue;
```

Reply:
If rampMetaData is null, it means there is no metadata for the ramp that was previously fetched. This shouldn’t happen, but it does for one ramp, which breaks the logic of creating quotes. I decided to skip this ramp for now.

Reply:
ok but like this you will skip all ramps that comes after the no metadata ramp, because you are using break which will go out of the for loop, instead we should use continue, am I right?

Reply:
I changed it, but please feel free to change it back if you think I am wrong, and let's discuss it